.github/workflows/auto.yml

```
@@ -0,0 +1,63 @@
+name: Automatic CI for TFE Modules
+
+on:
+  pull_request:
+    branches:
+      - master
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  terraform_format:
+
+    name: Run terraform fmt
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v1
+      with:
+        terraform_version: 0.14.7
+
+    - name: Format Public Install
+      run: |
+        terraform fmt -check -diff -recursive ${{ github.workspace }}
+
+  terraform_lint:
+
+    name: Run terraform-lint
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Terraform Lint
+      uses: terraform-linters/setup-tflint@v1
+      with:
+        tflint_version: v0.26.0
+
+    - name: Lint root module
+      run: |
+        tflint --config ${{ github.workspace }}/.tflint.hcl ${{ github.workspace }}
+
+    - name: Lint modules directory in a loop
+      run: |
+        for m in $(ls -1d modules/*/)
+        do
+          tflint \
+            --config ${{ github.workspace }}/.tflint.hcl \
+            ${{ github.workspace }}/${m}
+        done
+
+    - name: Lint tests directory in a loop
+      run: |
+        for m in $(ls -1d tests/*/)
+        do
+          tflint \
+            --config ${{ github.workspace }}/.tflint.hcl \
+            ${{ github.workspace }}/${m}
+        done
```

Let's make sure examples are also checked.

```suggestion
        done

      - name: Lint examples directory in a loop
        run: |
          for m in $(ls -1d examples/*/)
          do
            tflint \
              --config ${{ github.workspace }}/.tflint.hcl \
              ${{ github.workspace }}/${m}
          done
```

---

The examples directories contain no Terraform besides the personas which are being moved over to tests. I don't think we need to add examples yet.

---

If we do add the coverage now, it would ensure that future updates to the examples which add configuration immediately conform to the rest of the module. There are functional examples in the other modules which won't be part of the test case extraction so this is a consideration that we will need in the near future. What do you think?

---

I dig it. I'll add it.